container-search/src/main/java/com/yahoo/search/query/Sorting.java

```diff
@@ -426,20 +591,123 @@ public String toString() {
 
         @Override
         public int hashCode() {
-            return sortOrder.hashCode() + 17 * fieldSorter.hashCode();
+            return sortOrder.hashCode() + 17 * fieldSorter.hashCode() + 179 * missingValueSettings.hashCode();
         }
 
         @Override
         public boolean equals(Object o) {
             if (!(o instanceof FieldOrder other)) return false;
-            return other.sortOrder.equals(sortOrder) && other.fieldSorter.equals(fieldSorter);
+            return other.sortOrder.equals(sortOrder) && other.fieldSorter.equals(fieldSorter) &&
+                    other.missingValueSettings.equals(missingValueSettings);
         }
 
         @Override
         public FieldOrder clone() {
-            return new FieldOrder(fieldSorter.clone(), sortOrder);
+            return new FieldOrder(fieldSorter.clone(), sortOrder, missingValueSettings.getPolicy(),
+                    missingValueSettings.getMissingValue());
+        }
+
+    }
+
+    private static class Tokenizer {
```

Comment:
Consider removing `public` from all functions in this class. Some of them are already package private.